nemo/core/utils/numba_utils.py

```diff
@@ -148,6 +153,35 @@ def numba_cuda_is_supported(min_version: str) -> bool:
         return False
 
 
+def is_numba_cuda_fp16_supported(return_reason: bool = False) -> (bool, str):
```

Comment:
`-> Tuple[bool, str]`

`(bool, str)` is not a valid python type, and not supported.

Since `return_reason` changes the return type, the full type should be `Union[bool, Tuple[bool, str]]`